components/raftstore-v2/src/worker/tablet_gc.rs

```diff
@@ -107,27 +116,72 @@ impl<EK: KvEngine> Runner<EK> {
             logger,
             waiting_destroy_tasks: HashMap::default(),
             pending_destroy_tasks: Vec::new(),
+            background_pool: YatpPoolBuilder::new(DefaultTicker::default())
+                .name_prefix("tablet-gc-bg")
+                .thread_count(
+                    0,
+                    DEFAULT_BACKGROUND_POOL_SIZE,
+                    DEFAULT_BACKGROUND_POOL_SIZE,
+                )
+                .build_future_pool(),
         }
     }
 
-    fn trim(
-        tablet: &EK,
-        start_key: &[u8],
-        end_key: &[u8],
-        cb: Box<dyn FnOnce() + Send>,
-    ) -> engine_traits::Result<()> {
-        let start_key = keys::data_key(start_key);
-        let end_key = keys::data_end_key(end_key);
+    fn trim(&self, tablet: EK, start: Box<[u8]>, end: Box<[u8]>, cb: Box<dyn FnOnce() + Send>) {
+        let start_key = keys::data_key(&start);
+        let end_key = keys::data_end_key(&end);
         let range1 = Range::new(&[], &start_key);
         let range2 = Range::new(&end_key, keys::DATA_MAX_KEY);
-        tablet.delete_ranges_cfs(DeleteStrategy::DeleteFiles, &[range1, range2])?;
-        // TODO: Avoid this after compaction filter is ready.
-        tablet.delete_ranges_cfs(DeleteStrategy::DeleteByRange, &[range1, range2])?;
-        for r in [range1, range2] {
-            tablet.compact_range(Some(r.start_key), Some(r.end_key), false, 1)?;
+        // TODO: Avoid `DeleteByRange` after compaction filter is ready.
+        if let Err(e) = tablet
+            .delete_ranges_cfs(DeleteStrategy::DeleteFiles, &[range1, range2])
+            .and_then(|_| {
+                tablet.delete_ranges_cfs(DeleteStrategy::DeleteByRange, &[range1, range2])
+            })
+        {
+            error!(
+                self.logger,
+                "failed to trim tablet";
+                "start_key" => log_wrappers::Value::key(&start_key),
+                "end_key" => log_wrappers::Value::key(&end_key),
+                "err" => %e,
+            );
+            return;
         }
-        cb();
-        Ok(())
+        let logger = self.logger.clone();
+        self.background_pool
+            .spawn(async move {
```

Comment:
Is there anyway to track pending trim task count?  If not, maybe it's better to add a counter for this.

Reply:
Async pool has running tasks metrics. It is (sort of) equivalent to the pending tasks metrics of sync pool.